### ISSUE TITLE
[QUERY] How to use custom script extensions with Azure.ResourceManager.Compute?

### ISSUE THREAD

**Issue by guythetechie** [Azure/azure-sdk-for-net#13114]
**TLDR**
I'm at a loss on how to set [VirtualMachineScaleSetExtension.Setting](https://github.com/Azure/azure-sdk-for-net/blob/196f7e9afca5b6c576d19364aa30f57cc58a9db0/sdk/compute/Azure.ResourceManager.Compute/src/Generated/Models/VirtualMachineScaleSetExtension.cs#L63) in order to run custom script extensions.

**Query/Question**
I'm trying to create a scaleset with a custom script extension. The equivalent ARM template section would be:

```json
"extensionProfile": {
    "extensions": [
        {
            "name": "customScript",
            "properties": {
                "publisher": "Microsoft.Compute",
                "settings": {
                    "fileUris": [
                        "https://mysite.com/agent.exe"
                    ],
                    "commandToExecute": "agent.exe --parameter1 \"value\""
                },
                "typeHandlerVersion": "1.10",
                "autoUpgradeMinorVersion": true,
                "type": "CustomScriptExtension"
            }
        }
    ]
}
```
I can't seem to find a way to do something similar with the new Azure.ResourceManager.Compute library.
```csharp
var scaleSet = new VirtualMachineScaleSet(location)
{
    DoNotRunExtensionsOnOverprovisionedVMs = true,
    // ...
    VirtualMachineProfile = new VirtualMachineScaleSetVMProfile
    {
        ExtensionProfile = new VirtualMachineScaleSetExtensionProfile
        {
            Extensions = new List<VirtualMachineScaleSetExtension>
            {
                new VirtualMachineScaleSetExtension
                {
                    Name = "customScript",
                    Publisher = "Microsoft.Compute",
                    Settings = ??? // How do I set this?
                    TypeHandlerVersion = "1.10",
                    AutoUpgradeMinorVersion = true,
                    TypePropertiesType = "CustomScriptExtension"
                }
            }
        }
    }
};
```
I've tried using an anonymous object, but this fails with a **NotSupportedException**:
```csharp
Settings = new
{
    fileUris = new List<string>
    {
        fileUrl
    },
    commandToExecute = commandToExecute
}
```
I've also tried just passing a JSON string - the scaleset gets created, but the extension fails because the JSON isn't formatted properly. This code
```csharp
Settings = $"{{\"fileUris\": [\"{fileUrl}\"], \"commandToExecute\": \"{commandToExecute}\"}}"
```
returns this setting (note the extra quote the beginning and end)
![image](https://user-images.githubusercontent.com/31325620/86136595-486ce180-baba-11ea-849f-48e3927fe732.png)
as opposed to this desired setting
![image](https://user-images.githubusercontent.com/31325620/86136726-6b979100-baba-11ea-844b-150de718069c.png)

I looked at the SDK serialization code ([here](https://github.com/Azure/azure-sdk-for-net/blob/196f7e9afca5b6c576d19364aa30f57cc58a9db0/sdk/compute/Azure.ResourceManager.Compute/src/Generated/Models/VirtualMachineScaleSetExtension.Serialization.cs#L64) and [here](https://github.com/Azure/azure-sdk-for-net/blob/196f7e9afca5b6c576d19364aa30f57cc58a9db0/sdk/core/Azure.Core/src/Shared/AutoRest/Utf8JsonWriterExtensions.cs#L48)) and found that it does this, which explains my results:
```csharp
public static void WriteObjectValue(this Utf8JsonWriter writer, object value)
{
    switch (value)
    {
        //..
        case IUtf8JsonSerializable serializable:
            writer.WriteObjectValue(serializable);
            break;
        //..
        case string s:
            writer.WriteStringValue(s);
            break;
        //..
        default:
            throw new NotSupportedException("Not supported type " + value.GetType());
    }
}
```

So I tried writing a small CustomScriptSettings class that implements IUtf8JsonSerializable, but that interface is internal.

I understand this is still in preview, so this scenario might just not have been fully implemented yet.

Thanks!

**Environment:**
 - Name and version of the Library package used: Azure.ResourceManager.Compute 1.0.0-preview.1]

**Comment by erich-wang** [comment]
Thanks for raising the question, we're looking into it.

**Comment by guythetechie** [comment]
> Thanks for raising the question, we're looking into it.

Thank you. Any updates?

**Comment by guythetechie** [comment]
> > Thanks for raising the question, we're looking into it.
> 
> Thank you. Any updates?

Sorry, I accidentally closed this - reopening.

**Comment by keflem** [comment]
Hi @guythetechie!  I ran across this while searching for the exact same thing.  The answer isn't very intuitive, but if you use JObject you can get the Settings to work.  Example:

`  Settings = JObject.FromObject(new
   {
      fileUris = new string[] { scriptToRun },
      commandToExecute
  })
`

**Comment by allenjzhang** [comment]
@guythetechie , please let us know if above code sample does not work for you. Thanks.
